pkg/api/v1alpha1/cloudstackdatacenterconfig_webhook.go

```diff
@@ -86,10 +87,10 @@ func validateImmutableFieldsCloudStackCluster(new, old *CloudStackDatacenterConf
 		)
 	}
 
-	if old.Spec.Zone != new.Spec.Zone {
+	if !reflect.DeepEqual(old.Spec.Zones, new.Spec.Zones) {
```

Comment:
We have been trying to avoid using deep equal aside from in tests
I would suggest creating an `Equal` to compare `[]Zone`
I think there already other api structs that implement this pattern

Reply:
Equal could not be applied directly to []Zone, I added one to apply Zone